client/app/scripts/components/nodes.js

```diff
@@ -73,7 +73,15 @@ class Nodes extends React.Component {
   }
 
   setDimensions() {
-    this.props.setViewportDimensions(window.innerWidth, window.innerHeight);
+    // Use the `scope-app` div to determine the viewport rectangle, because `window` gives
+    // wrong values for scope container inside Weave Cloud (on dev & prod), because of the
+    // top navigation toolbar that is counted in, even though it's not part of Scope.
+    const viewport = document.getElementsByClassName('scope-app')[0].getBoundingClientRect();
```

Comment:
bit icky to reach up the cmp hierarchy to grab `scope-app`, maybe move `setDimensions` up to `App`?